src/models/preHooks/validate.js

```diff
@@ -14,6 +15,23 @@ module.exports = {
     if (!hasCompany && !isPopulate && !isVendorUser && !requestingOwnInfos && !allCompanies) next(Boom.badRequest());
     next();
   },
+  formatQuery(next) {
+    const query = this.getQuery();
+    const flattenQuery = {};
+    for (const entry of Object.entries(query)) {
+      if (entry[1] && typeof entry[1] === 'object' && !Array.isArray(entry[1]) && Object.keys(entry[1])[0] &&
+        Object.keys(entry[1])[0].charAt(0) !== '$') {
```

Comment:
que cherches tu a faire la ?

Reply:
j'ai modifié un peu la condition avec le retour de kenny mais en gros l'idée c'est de flat si on a un un nestedpath du style `image: { link: 'https://alenvi.io', publicId: 'abcdefgh' }` pour avoir quelque chose du genre `'image.link': 'https://alenvi.io', 'image.publicId':  'abcdefgh'` mais le truc c'est que quand on a des opérateurs style $in, $set, $or c'est aussi dans un objet nesté et dans ce cas là on veut pas flat donc la condition dit : 
si la valeur existe et est un objet qui n'est pas un tableau (car typeof d'un tableau c'est object) et que la première key dans cet objet nesté n'est pas un opérateur commençant par $ alors on flat

Reply:
est ce que tu peux essayer de rendre ce plus explicite en nommant les variables par exemple ?

Reply:
et as tu besoin des 2 conditions : `typeof entry[1] === 'object'` `!Array.isArray(entry[1])` ? j'ai l'impression que si la premiere est vraie, la 2eme l'est aussi et si la 1ere est fausse alors on ne passe pas pas la seconde

Reply:
justement non, comme je disais un tableau est considéré comme un objet par typeof donc on risque de rentrer dans la condition si on a un tableau alors qu'on veut pas. Tu peux tester ça fait péter pas mal de tests.
Concernant le fait de mettre dans des variables (je suppose que tu parlais de Object.keys(value)[0]), le problème est que
si value pas défini : 
```
const nestedObjectFirstKey = Object.keys(value)[0];
      if (value && typeof value === 'object' && !Array.isArray(value) && nestedObjectFirstKey &&
      nestedObjectFirstKey.charAt(0) !== '$') 
```
va péter au moment de définir nestedObjectFirstKey

alors qu'avec 
```
      if (value && typeof value === 'object' && !Array.isArray(value) && Object.keys(value)[0] &&
      Object.keys(value)[0].charAt(0) !== '$')
```
ça pète pas vu que si value n'est pas défini on quitte direct la condition sans tester Object.keys(value)[0]

Reply:
Que dis-tu de séparer la condition en plusieurs étapes comme : 
```
for (const [key, value] of Object.entries(query)) {
      const isObject = value && typeof val === 'object'  && !Array.isArray(value);
      const valueHasNotDollarSign = Object.keys(value)[0] && Object.keys(value)[0].charAt(0) !== '$';
      if (isObject && valueHasNotDollarSign) {
        const flattenEntry = flat({ [key]: value });
        Object.assign(flattenQuery, flattenEntry);
      } else {
        const untouchedEntry = { [key]: value };
        Object.assign(flattenQuery, untouchedEntry);
      }
```

Reply:
ça continue de casser si jamais value n'est pas défini comme je l'expliquais au-dessus mais je peux rajouter value dans la variable valueHasNotDollarSign pour que ça passe